gravitee-am-ui/src/app/domain/settings/password-policy/domain-password-policy.component.ts

```diff
@@ -132,12 +133,38 @@ export class DomainPasswordPolicyComponent implements OnInit {
       passwordHistoryEnabled: this.passwordHistoryEnabled,
       oldPasswords: this.oldPasswords,
     };
-    this.domainService.patchPasswordSettings(this.domainId, data).subscribe((data) => {
-      this.passwordSettings = data.passwordSettings;
+    this.domainService.patchPasswordSettings(this.domainId, data).subscribe((response) => {
+      this.passwordSettings = response.passwordSettings;
       this.domain['passwordSettings'] = this.passwordSettings;
       this.form.reset(this.passwordSettings);
       this.formChanged = false;
       this.snackbarService.open('Password settings configuration updated');
     });
   }
+  confirmResetDialog(event: any): void {
+    event.preventDefault();
+    this.dialogService
+      .confirm('Reset password policy', 'Are you sure you want to reset your password policy settings and use the default one?')
+      .subscribe((res) => {
+        if (res) {
+          this.reset();
+        }
+      });
+  }
+  reset(): void {
+    this.domainService.patchPasswordSettings(this.domainId, { passwordSettings: { inherited: true } }).subscribe(() => {
```

Comment:
**todo**: based on the comment on gravitee-am-service/src/main/java/io/gravitee/am/service/model/PatchDomain.java we should set passwordSettings to null.

```suggestion
    this.domainService.patchPasswordSettings(this.domainId, { passwordSettings: null }).subscribe(() => {
```